RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripRawToClusterAlgo.h

```diff
@@ -0,0 +1,221 @@
+
+#include "DataFormats/Portable/interface/PortableHostCollection.h"
+#include "DataFormats/Portable/interface/alpaka/PortableCollection.h"
+
+#include "DataFormats/SiStripClusterSoA/interface/alpaka/SiStripClusterDevice.h"
+#include "DataFormats/SiStripDigiSoA/interface/alpaka/SiStripDigiDevice.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "RecoLocalTracker/SiStripClusterizer/interface/alpaka/SiStripClusterizerConditionsDevice.h"
+#include "RecoLocalTracker/SiStripClusterizer/interface/alpaka/SiStripMappingDevice.h"
+
+namespace edm {
+  class ParameterSet;
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// The StripClustersAuxSoALayout is an auxiliary SoA, with the same size of the SiStripDigiSoA, which helps
+// in the clusterization process.
+namespace sistrip {
+  GENERATE_SOA_LAYOUT(StripClustersAuxSoALayout,
+                      SOA_COLUMN(uint32_t, seedStripsMask),
+                      SOA_COLUMN(uint32_t, seedStripsNCMask),
+                      SOA_COLUMN(uint32_t, seedStripsNCIndex),
+                      SOA_COLUMN(uint32_t, prefixSeedStripsNCMask),
+                      //
+                      SOA_SCALAR(float, channelThreshold),
+                      SOA_SCALAR(float, seedThreshold),
+                      SOA_SCALAR(float, clusterThresholdSquared),
+                      SOA_SCALAR(uint8_t, maxSequentialHoles),
+                      SOA_SCALAR(uint8_t, maxSequentialBad),
+                      SOA_SCALAR(uint8_t, maxAdjacentBad),
+                      SOA_SCALAR(float, minGoodCharge),
+                      SOA_SCALAR(uint16_t, clusterSizeLimit))
+
+  using StripClustersAuxSoA = StripClustersAuxSoALayout<>;
+  using StripClustersAuxView = StripClustersAuxSoA::View;
+  using StripClustersAuxConstView = StripClustersAuxSoA::ConstView;
+  using StripClustersAuxHost = PortableHostCollection<StripClustersAuxSoA>;
+}  // namespace sistrip
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip {
+  using namespace ::sistrip;
+  using StripClustersAuxDevice = PortableCollection<StripClustersAuxSoA>;
+}  // namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip
+
+ASSERT_DEVICE_MATCHES_HOST_COLLECTION(sistrip::StripClustersAuxDevice, sistrip::StripClustersAuxHost);
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip {
+  struct FEDChMetadata {
+    uint32_t detId;
+    uint16_t fedId;
+    uint8_t fedCh;
+    //
+    uint16_t fedChOfs;
+    uint32_t fedChOfs_wrt_rawFedId;
+    uint8_t fedBufferRO;
+  };
+
+  namespace fedChannelDetails {
+    // It corresponds to the one in EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h,
+    // but for Alpaka, with the addition of stripsInCh function
+    class FEDChannel {
```

Comment:
Instead of du0licating the code from EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h, would it be feasible to
  - extend the code in EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h to add the treatment of lite/non-lite channels
  - mark all functions as `constexpr` instead of `ALPAKA_FN_HOST_ACC`, to make them available in device code

?